src/main/java/com/societegenerale/commons/plugin/rules/ConstantNamesRuleTest.java

```diff
@@ -0,0 +1,104 @@
+package com.societegenerale.commons.plugin.rules;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.fields;
+
+import java.util.Collection;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.societegenerale.commons.plugin.service.ScopePathProvider;
+import com.societegenerale.commons.plugin.utils.ArchUtils;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.domain.JavaEnumConstant;
+import com.tngtech.archunit.core.domain.JavaField;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+
+/**
+ * 
+ * Constant names and Enum values have to be written in uppercase.
+ * 
+ * It's possible to add underscore but not at the beginning or the end of the
+ * name.
+ * 
+ * @see <a href= "https://rules.sonarsource.com/java/RSPEC-115">Constant names
+ *      should comply with a naming convention</a>
+ * 
+ * 
+ */
+
+public class ConstantNamesRuleTest implements ArchRuleTest {
+
+	private static Pattern pattern;
```

Comment:
why making these static ?

Reply:
Because of the static context due to **protected static ArchCondition** method used by **should**.

If pattern is not static then this error occurs : 

*Cannot make a static reference to the non-static field pattern.*

*1 quick fix available:*

*Change 'pattern' to 'static'*

Same thing with matcher.